src/Swift.Bindings/src/Emitter/StringEmitter/Handler/TypeHandler.cs

```
@@ -588,28 +505,46 @@ private void EmitPrivateConstructor()
         /// </summary>
         private void WriteMarshalToSwiftFrozenStruct()
         {
-            string payloadName = "this";
             TypeRecord typeRecord = _typeDatabase.GetTypeRecordOrThrow(_structDecl.SwiftTypeName);
             if (MarshallingHelpers.IsFrozenStructProjectedAsClass(typeRecord))
             {
-                payloadName = "_payload";
-            }
+                var text = $$"""
+                unsafe void ISwiftObject.MarshalToSwift(Span<byte> swiftDestSpan)
+                {
+                    var metadata = SwiftObjectHelper<{{_structDecl.Name}}>.GetTypeMetadata();
+                    Debug.Assert((int)metadata.Size == swiftDestSpan.Length, $"Span size does not match type size, Expected: {(int)metadata.Size}, Actual: {swiftDestSpan.Length}");
+                    fixed (void* swiftDest = swiftDestSpan)
+                    {
+                        // Ensure that the instance is valid before making copy
+                        bool success = false;
+                        _payload.DangerousAddRef(ref success);
+                        metadata.ValueWitnessTable->InitializeWithCopy((void *)swiftDest, _payload, metadata);
+                        if (success)
+                            _payload.DangerousRelease();
```

I think this needs a try/finally to make sure the payload is released if an exception occurs.

---

So my understanding here is `InitializeWIthCopy` is an interop call (calling some Swift API). If that is true, then try/finally block isn't needed since it will be a fail fast at the interop boundary. Same for the rest of these. We shouldn't be wrapping interop calls with try/finally because they will never be run.